src/Web3.elm

```diff
@@ -186,17 +190,133 @@ toChecksumAddress (Address address) =
         }
 
 
+toWei : EthUnit -> String -> Result String BigInt
+toWei unit amount =
+    -- check to make sure input string is formatted correctly, should never error in here.
+    if Regex.contains (Regex.regex "^\\d*\\.?\\d+$") amount then
+        let
+            decimalPoints =
+                decimalShift unit
 
-{-
-   TODO
-   fromWei : EthUnit -> BigInt -> String
+            formatMantissa =
+                String.slice 0 decimalPoints >> String.padRight decimalPoints '0'
 
-   toWei : EthUnit -> String -> Maybe BigInt
+            finalResult =
+                case (String.split "." amount) of
+                    [ a, b ] ->
+                        a ++ (formatMantissa b)
 
-   bigIntToWei : EthUnit -> BigInt -> BigInt
+                    [ a ] ->
+                        a ++ (formatMantissa "")
 
--}
--- CORE
+                    _ ->
+                        "ImpossibleError"
+        in
+            case (BigInt.fromString finalResult) of
+                Just result ->
+                    Ok result
+
+                Nothing ->
+                    Err "There was an error calculating the result. However, the fault is not yours; please report this bug on github."
+    else
+        Err "Malformed number string passed to `toWei` function."
+
+
+fromWei : EthUnit -> BigInt -> String
+fromWei unit amount =
+    let
+        decimalIndex =
+            decimalShift unit
+
+        -- There are under 10^27 wei in existance (so we safe for the next couple of malenium of mining).
+        amountStr =
+            BigInt.toString amount |> String.padLeft 27 '0'
+
+        result =
+            (String.left (27 - decimalIndex) amountStr)
+                ++ "."
+                ++ (String.right decimalIndex amountStr)
+    in
+        result
+            |> Regex.replace Regex.All
+                (Regex.regex "(^0*(?=0\\.|[1-9]))|(\\.?0*$)")
```

Comment:
The only potentially confusing line of my code. Essentially it turns "000.0123" -> "0.0123"; "00012.3" -> "12.3"; "0001230.000" -> "1230"; "00012.3000" -> "12.3"

Reply:
Good to know. I'm assuming your code is putting the leading zeros on in the first place, as part of it's magic?